### DISCUSSION TITLE
Set a specific display name for a registered model

### DISCUSSION THREAD

**Discussion by seanlinsley** [activeadmin/activeadmin#7884] [Ideas]
The point of this is to allow fancy display names without requiring that ActiveRecord instantiate huge numbers of records, which is possible for example on the default belongs_to form that AA creates, which adds a dropdown to choose a parent.

The simple way to get around this is to only pluck the desired DB attribute, saving a huge amount of time:

``` ruby
User.pluck :username
```

But that's not always possible, because sometimes you want to combine attributes to create a special string:

``` ruby
User.select([:first_name,:last_name]).map &:full_name
```

Right now, though, we just do this:

``` ruby
User.all.map &:full_name
```

I'm not sure what the syntax should be, or where it should be put, but I'm very much of the opinion that you should be able to declare both what display method you want used, as well as what DB attributes are required.

Maybe something like this:

``` ruby
# Just plucks directly from the DB. This is the fastest option
display_with :username, simple: true

# Calls the full_name method on the model, with the requested attrs
display_with :full_name, requires: [:first_name, :last_name]
```

I'm not sure what a good syntax would be for decorators, since I haven't used them much.

**Comment by timoschilling** [comment]
One other option is:

``` ruby
display_with "CONCAT(`firstname`,' ',`lastname`)"
```

This based on:

``` ruby
User.pluck("CONCAT(`firstname`,' ',`lastname`)")
```

**Comment by timoschilling** [comment]
related to #2166 and justinfrench/formtastic#1085

**Comment by Andrew-Max** [comment]
This would be incredibly helpful :+1:
